subsys/fb/cfb.c

```diff
@@ -440,30 +544,95 @@ int cfb_framebuffer_invert(const struct device *dev)
 	return 0;
 }
 
-int cfb_framebuffer_finalize(const struct device *dev)
+int cfb_framebuffer_finalize_area(const struct device *dev, uint16_t left, uint16_t top,
+				  uint16_t width, uint16_t height)
 {
 	const struct display_driver_api *api = dev->api;
 	const struct char_framebuffer *fb = &char_fb;
+#ifdef CONFIG_CHARACTER_FRAMEBUFFER_TRANSFER_WITH_PIXEL_FORMAT_CONVERSION
+	const uint8_t bpp = cfb_bytes_per_pixel(fb->pixel_format);
+#endif
 	struct display_buffer_descriptor desc;
 	int err;
 
 	if (!fb || !fb->buf) {
 		return -ENODEV;
 	}
 
-	desc.buf_size = fb->size;
-	desc.width = fb->x_res;
-	desc.height = fb->y_res;
-	desc.pitch = fb->x_res;
+	if ((fb->pixel_format == PIXEL_FORMAT_MONO01 || fb->pixel_format == PIXEL_FORMAT_MONO10) &&
+	    (fb->screen_info & SCREEN_INFO_MONO_VTILED)) {
+		/* A VTILED format display usually does not support partial region transfers.
+		 * Transfer the entire framebuffer.
+		 */
+		desc.buf_size = fb->size;
+		desc.width = fb->x_res;
+		desc.height = fb->y_res;
+		desc.pitch = fb->x_res;
+
+		if (!(fb->pixel_format & PIXEL_FORMAT_MONO10) != !(fb->inverted)) {
+			cfb_invert(fb);
+			err = api->write(dev, 0, 0, &desc, fb->buf);
+			cfb_invert(fb);
+			return err;
+		}
+
+		return api->write(dev, 0, 0, &desc, fb->buf);
+	}
+
+#ifdef CONFIG_CHARACTER_FRAMEBUFFER_TRANSFER_WITH_PIXEL_FORMAT_CONVERSION
```

Comment:
```c
if(ENABLED(CFB_TRANSFER_WITH_PIXEL_FORMAT_CONVERSION)) {
	cfb_transfer_with_pixel_conversion(..);
}

```
I do not understand why this code should not always be built, and why it cannot be broken into smaller chunks. As it is now, it is difficult to understand and review.